src/explorer/trialApp/TrialAppTreeItem.ts

```diff
@@ -79,7 +82,7 @@ export class TrialAppTreeItem extends SiteTreeItemBase implements ISiteTreeItem
     }
 
     public async refreshImpl(): Promise<void> {
-        this.metadata = await TrialAppTreeItem.getTrialAppMetaData(this.metadata.loginSession);
+        this.client.metadata = await TrialAppClient.getTrialAppMetaData(this.metadata.loginSession);
```

Comment:
I think it would be better to completely re-create the client instead of just setting the metadata. For example, if `publishingPassword` changed, we want to pick up that change when they refresh - and it's only set in the constructor of `TrialAppClient`.

At that point, I might just move all of the code from `getTrialAppMetaData` into `createTrialAppClient` and delete  `getTrialAppMetaData` since I think it's only used once.